src/utils/AuthProvider.tsx

```diff
@@ -0,0 +1,126 @@
+'use client';
+
+import { AuthError, AuthResponse, Session } from '@supabase/supabase-js';
+import {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import type { UUID } from 'crypto';
+import supabase from '@/api/supabase/createClient';
+
+export interface AuthContextType {
+  session?: Session;
+  userId: UUID | undefined;
+  userEmail: string | undefined;
+  signIn: (email: string, password: string) => Promise<AuthResponse>;
+  signOut: () => Promise<AuthError | null>;
+  signUp: (
+    email: string,
+    password: string,
+    options: object,
```

Comment:
nit: let's use `SignUpOptions` here

Reply:
also thoughts on consolidating these miscellaneous types into a `@/types/misc.d.ts` file? (cc @varortz )

Reply:
would this also include types like `OnboardingContextType`, `ProfileContextType`?

Reply:
No, I think context types are best left in the relevant context files. In hindsight, it may be better to leave the `SignUpOptions` type here since we don't need to reference the type anywhere else as well.

Reply:
Yeah, if it isn't used somewhere else probably best to leave it in this file.